Makefile

```diff
@@ -0,0 +1,64 @@
+	cd build && cmake -DDEBUG=ON -DDEVELOPER_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_COAP_PAYLOAD=ON -DTRACE_LEVEL=DEBUG -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-developer: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DDEVELOPER_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_COAP_PAYLOAD=ON -DTRACE_LEVEL=INFO -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-developer-thread: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DENABLE_THREAD_SANITIZE=1 -DDEVELOPER_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_COAP_PAYLOAD=ON -DTRACE_LEVEL=ERROR -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-developer-with-coverage: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DENABLE_COVERAGE=1 -DDEVELOPER_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_COAP_PAYLOAD=ON -DTRACE_LEVEL=INFO -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-byoc-with-coverage: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DENABLE_COVERAGE=1 -DBYOC_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_COAP_PAYLOAD=ON -DTRACE_LEVEL=INFO -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-byoc: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DBYOC_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_LEVEL=INFO -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-byoc-thread: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DENABLE_THREAD_SANITIZE=1 -DBYOC_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_LEVEL=INFO -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-byoc-debug: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DBYOC_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_LEVEL=DEBUG -DTRACE_COAP_PAYLOAD=ON -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-byoc-debug-udp: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DBYOC_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_LEVEL=DEBUG -DTRACE_COAP_PAYLOAD=ON -DCMAKE_BUILD_TYPE=Debug -DCLOUD_CLIENT_CONFIG=../config/mbed_cloud_client_udp_user_config.h .. && make -j ${JOBS} && cd ..
+
+build-factory: lib/mbed-cloud-client/source/update_default_resources.c generate-cli-parsers build
+	cd build && cmake -DFACTORY_MODE=ON -DFIRMWARE_UPDATE=ON -DTRACE_LEVEL=INFO -DCMAKE_BUILD_TYPE=Debug .. && make -j ${JOBS} && cd ..
+
+build-doc:
+	mkdir -p build-doc && cd build-doc && cmake -DBUILD_DOCUMENTATION=ON .. && make edge-doc
+	echo "\033[0;33mDocumentation is at ./build-doc/doxygen/index.html\033[0m"
+
+run-edge-core:
+	./build/bin/edge-core -p 22225 -o 8080
+
+run-edge-core-valgrind:
```

Comment:
Please remove this. Osmo tests are not public

Reply:
I have removed that section.